### PR TITLE
Update lib.rs

### PR DESCRIPTION
**Pull request by ndkazu**
In the pallet roles:
- Added descriptive Events and Errors for the different role/account creation
- Added a Seller/Servicer account creation rejection function for the admin
- Added descriptives events in Account_creation approval & rejection functions
- Added additional check in Account_creation approval & rejection functions (caller account is not account to be verified)

### CODE REVIEW

File: pallets/roles/src/lib.rs [hunk 1]
```diff
@@ -85,7 +78,15 @@ pub mod pallet {
...
-		SomethingStored(u32, T::AccountId),
+		
+		InvestorCreated(<T as frame_system::Config>::BlockNumber,<T as frame_system::Config>::AccountId),
+		TenantCreated(<T as frame_system::Config>::BlockNumber,<T as frame_system::Config>::AccountId),
+		SellerCreated(<T as frame_system::Config>::BlockNumber,<T as frame_system::Config>::AccountId),
+		ServicerCreated(<T as frame_system::Config>::BlockNumber,<T as frame_system::Config>::AccountId),
+		AccountCreationApproved(<T as frame_system::Config>::BlockNumber,<T as frame_system::Config>::AccountId),
+		SellerAccountCreationRejected(<T as frame_system::Config>::BlockNumber,<T as frame_system::Config>::AccountId),
+		ServicerAccountCreationRejected(<T as frame_system::Config>::BlockNumber,<T as frame_system::Config>::AccountId),
+		CreationRequestCreated(<T as frame_system::Config>::BlockNumber,<T as frame_system::Config>::AccountId),
```
Comment (ilhanu) [inline pallets/roles/src/lib.rs:89]: Why are the Events typed as selected? If they are a part of the `frame_system` and `	#[pallet::config]` can they not be typed as as following 

example: `		InvestorCreated(blocknumber T::BlockNumber, accountId T::AccountId)`

File: pallets/roles/src/lib.rs [hunk 2]
```diff
@@ -105,82 +110,141 @@ pub mod pallet {
...
+	  #[pallet::weight(10_000 + T::DbWeight::get().writes(1))]
+      ///Approval function for Sellers and Servicers. Only for admin level.
+      pub fn account_approval(origin:OriginFor<T>,account: T::AccountId)-> DispatchResult{
+         ensure_root(origin.clone())?;
+		 let caller = ensure_signed(origin)?;
+		 ensure!(caller.clone()!=account.clone(),Error::<T>::InvalidOperation);
+         Self::approve_account(account)?;
+		 let now = <frame_system::Pallet<T>>::block_number();
+		 Self::deposit_event(Event::AccountCreationApproved(now,caller));
+         Ok(().into())
```
Comment (ilhanu) [inline pallets/roles/src/lib.rs:172]: Nice addition 👍🏼 

We can also give the role a name like a [membership-manager](https://github.com/paritytech/substrate/blob/bc310fdd1040af8a3d2901b5d6f4cf8d5250aeb7/frame/alliance/src/lib.rs#L238-L239)

**Review by ilhanu — CHANGES_REQUESTED** [review]
Thanks a lot for the work, it looks really neat! I'm at the moment heavily inspired by the alliance pallet as you can read in the review, it has bits of similar work and nice coding styles. We don't necessarily need to adhere to those standards, however it will make our code easier readable and future-proof.

Also it would be nice if we could make every PR standard especially when adding features that we also add tests to it. @MrishoLukamba was working on it in the #40, not sure what the plan is with that PR, but it would be nice before merging this that we have some basic unit tests that can test these functions.

File: pallets/roles/src/lib.rs (hunk 1 above)
Comment (ndkazu) [reply pallets/roles/src/lib.rs:89]: You're totally right. I blindly mimicked the parity node-template syntax, but it's really heavy.

**Comment by MrishoLukamba** [comment]
done the testing on main

**Review by letodunc — APPROVED** [review]

File: pallets/roles/src/lib.rs [hunk 3]
```diff
@@ -79,14 +79,14 @@ pub mod pallet {
...
-		ServicerAccountCreationRejected(<T as frame_system::Config>::BlockNumber,<T as frame_system::Config>::AccountId),
-		CreationRequestCreated(<T as frame_system::Config>::BlockNumber,<T as frame_system::Config>::AccountId),
+		InvestorCreated(T::BlockNumber,T::AccountId),
+		TenantCreated(T::BlockNumber,T::AccountId),
+		SellerCreated(T::BlockNumber,T::AccountId),
+		ServicerCreated(T::BlockNumber,T::AccountId),
+		AccountCreationApproved(T::BlockNumber,T::AccountId),
+		SellerAccountCreationRejected(T::BlockNumber,T::AccountId),
+		ServicerAccountCreationRejected(T::BlockNumber,T::AccountId),
+		CreationRequestCreated(T::BlockNumber,T::AccountId),
```
Comment (ilhanu) [inline pallets/roles/src/lib.rs:90]: Sweet!

File: pallets/roles/src/lib.rs [hunk 4]
```diff
@@ -184,8 +173,22 @@ pub mod pallet {
...
 	  }
 
+	  #[pallet::weight(10_000 + T::DbWeight::get().writes(1))]
+	  ///The caller will transfer his manager authority to a different account
+	  pub fn set_manager(origin:OriginFor<T>,new: <T::Lookup as StaticLookup>::Source)->DispatchResult{
+		//ensure_signed(origin.clone())?;
+		ensure_root(origin.clone())?;
+		SUDO::Pallet::<T>::set_key(origin,new).ok();
+		Ok(().into())
+	}
```
Comment (ilhanu) [inline pallets/roles/src/lib.rs:183]: Yeah this is a good start for the mananager or the verifier we called as in the issue #36

**Review by ilhanu — APPROVED** [review]
Thanks for the improvements, the last two commits could have also been handled with the following syntax
```
git checkout main
git merge origin/37_housing_fund (example) 
```
It make it much more understandable where the code comes from. You can squash and merge it.

**Pull request merged**